pandas/core/generic.py

```diff
@@ -3455,7 +3455,7 @@ def replace(self, to_replace=None, value=None, inplace=False, limit=None,
         ----------
         method : {'linear', 'time', 'index', 'values', 'nearest', 'zero',
                   'slinear', 'quadratic', 'cubic', 'barycentric', 'krogh',
-                  'polynomial', 'spline' 'piecewise_polynomial', 'pchip',
+                  'polynomial', 'spline' 'from_derivatives', 'pchip',
```

Comment:
we need to leave the original method ``piecewise_polynomial`` and add the new one ``from_derivatives``. Then switch using the version of scipy. E.g. if > 0.17 and ``piecewise_polynomila`` -> ``from_derivatives``. If passed ``from_derivatives`` I would just use it.